src/views/virtualmachines/migrate/VirtualMachineMigration.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { FC, useState } from 'react';
+import { Trans } from 'react-i18next';
+import { useNavigate, useParams } from 'react-router-dom-v5-compat';
+
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import InlineFilterSelect from '@kubevirt-utils/components/FilterSelect/InlineFilterSelect';
+import Loading from '@kubevirt-utils/components/Loading/Loading';
+import useDefaultStorageClass from '@kubevirt-utils/hooks/useDefaultStorage/useDefaultStorageClass';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import {
+  modelToGroupVersionKind,
+  StorageClassModel,
+  VirtualMachineModelGroupVersionKind,
+} from '@kubevirt-utils/models';
+import { getName } from '@kubevirt-utils/resources/shared';
+import useDisksSources from '@kubevirt-utils/resources/vm/hooks/disk/useDisksSources';
+import { ResourceLink, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Alert,
+  AlertVariant,
+  Bullseye,
+  Stack,
+  StackItem,
+  Title,
+  Wizard,
+  WizardHeader,
+  WizardStep,
+} from '@patternfly/react-core';
+import { Table, Tbody, Td, Tr } from '@patternfly/react-table';
+
+import { migrateVM } from './utils';
+
+const StorageClassModelGroupVersionKind = modelToGroupVersionKind(StorageClassModel);
+
+const VirtualMachineMigrate: FC = () => {
+  const { t } = useKubevirtTranslation();
+
+  const { name, namespace } = useParams<{ name: string; namespace: string }>();
+  const [selectedStorageClass, setSelectedStorageClass] = useState('');
+  const [migrationError, setMigrationError] = useState<Error | null>(null);
+  const [migrationLoading, setMigrationLoading] = useState(false);
+
+  const navigate = useNavigate();
+
+  const [vm, loaded, error] = useK8sWatchResource<V1VirtualMachine>(
+    name
+      ? {
+          groupVersionKind: VirtualMachineModelGroupVersionKind,
+          name,
+          namespace,
+        }
+      : null,
+  );
+
+  const { pvcs } = useDisksSources(vm);
+
+  const [{ clusterDefaultStorageClass, sortedStorageClasses }, scLoaded] = useDefaultStorageClass();
+
+  const defaultStorageClassName = getName(clusterDefaultStorageClass);
+
+  const destinationStorageClass = selectedStorageClass || defaultStorageClassName;
+
+  const onSubmit = async () => {
+    setMigrationLoading(true);
+    setMigrationError(null);
+    try {
+      await migrateVM(vm, pvcs, destinationStorageClass);
+
+      navigate(-1);
+    } catch (apiError) {
+      setMigrationError(apiError);
+    }
+
+    setMigrationLoading(false);
+  };
+
+  if (!loaded && !scLoaded)
+    return (
+      <Bullseye>
+        <Loading />
+      </Bullseye>
+    );
+
+  if (error)
+    return (
+      <Alert isInline title={t('Error')} variant={AlertVariant.danger}>
+        {error?.message}
+      </Alert>
+    );
```

Comment:
We should create a generic HOC that gets loaded and error, and return an alert or loader or children accordingly instead of repeating snippets around the app. Could you please open a ticket for it?

Reply:
I'll do that in a different pr

Reply:
i see u open a ticket, thank you